### ISSUE TITLE
No transports returned in negotiation response

### ISSUE THREAD

**Issue by siberianguy** [aspnet/SignalR#1220]
I'm trying to connect to SignalR server that is hidden behind CloudFlare. To do that I first make a request using CloudFlareUtilities (it solves js problem) to retrieve a cookie. After that, I specify that cookie and the same Http-Agent to SignalR. The problem is I get a FormatException with a message specified in the title. I'm confused with this error message. So it seems like there's a SignalR server by a specified url but somehow negotiation doesn't go successfully. Can it be relate to the fact that the server is using a full .NET version of SignalR?

```
            const string userAgent = "SignalR.Client.NET45/2.2.2.0 (Microsoft Windows NT 6.2.9200.0)";

            var requestMessage = new HttpRequestMessage(HttpMethod.Get, "https://bittrex.com");
            requestMessage.Headers.TryAddWithoutValidation("User-Agent", userAgent);

            var cookieContainer = new CookieContainer();
            var client = new HttpClient(new ClearanceHandler(new HttpClientHandler
            {
                CookieContainer = cookieContainer
            }));

            var response = await client.SendAsync(requestMessage);

            var hubConnection = new HubConnectionBuilder()
                .WithUrl("https://socket.bittrex.com/signalr/Core")
                .WithConsoleLogger(LogLevel.Trace)
                .WithWebSocketOptions(options =>
                {
                    options.Cookies = cookieContainer;
                    options.SetRequestHeader("User-Agent", userAgent);
                })
                .Build();

            await hubConnection.StartAsync();
```

**Comment by BrennanConroy** [comment]
You're using SignalR 2.X on the server and not SignalR Core? That's not supported and will never be supported.
The new SignalR Core is a complete rewrite and not compatible with old SignalR

**Comment by siberianguy** [comment]
What is the reasoning behind this design decision? Just imagine if .NET Core applications did not support communicating with REST APIs written in full .NET Framework. Sounds ridiculous, right? I understand, it's a rewrite but supporting SignalR protocol sounds like a reasonable feature. So basically, you're saying there's no and will no be a way to communicate with old SignalR server using .NET Core, right?

**Comment by davidfowl** [comment]
That’s correct. The decision was made so that we could improve the protocol to make it more efficient. To support things like:

- Direct connections via websockets 
- Faster connection times (less http requests per connection on startup)
- supports non http transports (tcp)
- The new protocol also supports binary data
- The protocol is finally pluggable (this is a large architectural change)
- Support for volatile messaging (this was pretty impossible with he old design)

It’s not a .net Core vs .net Framework thing, the new client supports .net standard which runs on both. The new SignalR is incompatible with the old one.

**Comment by siberianguy** [comment]
@davidfowl, those are all good reasons but this decision would look much better if the old SignalR .NET client was refactored to support .NET Standard, so those who need to communicate with thousands of already existing SignalR servers were not pushed to use full .NET Framework.

**Comment by davidfowl** [comment]
It always does. Not sure what you mean.

**Comment by siberianguy** [comment]
Sorry, my fault. I got lost in the clients. The problem I have is tricky. Old client doesn't support specifying User-Agent header which is critical if you want to bypass CloudFlare. The solution community came to is to use WebSocket-Sharp for transport: https://github.com/elcattivo/CloudFlareUtilities/issues/18 And the problem is WebSocket-Shart doesn't support .NET Core, so we're stuck with full .NET framework there. I tried to overcome this problem by using a new version of SignalR client that does support User-Agent header manipulation but it turned out the new client doesn't work with the old server... Just not my day :)
